mailbox/api/src/test/java/org/apache/james/mailbox/SubscriptionManagerContract.java

```diff
@@ -0,0 +1,91 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.exception.SubscriptionException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+public interface SubscriptionManagerContract {
+
+    Username USER1 = Username.of("test");
+    String MAILBOX1 = "test1";
+    String MAILBOX2 = "test2";
+    MailboxSession SESSION = MailboxSessionUtil.create(USER1);
+
+    SubscriptionManager getSubscriptionManager();
+
+    @AfterEach
+    default void teardown() throws SubscriptionException {
```

Comment:
Can we have a commit for getting rid of this aweful lifecycle management method in a contract? <3

Cassandra + memory have this lifecycle comming for free, all you need is clear the JPA tables...

Reply:
I thought I removed that... but I had to do a rebase before pushing, maybe it got messed up. Will fix